### ISSUE TITLE
bs-modal does not close anymore

### ISSUE THREAD

**Issue by iamsebastian** [kaliber5/ember-bootstrap#155]
Hello people,

Since I've upgraded to `0.11.1` from `0.9.1`, the `bs-modal` does not close anymore. Not by clicking outside the modal nor the default close button nor the `x` in the corner.

The modal is implemented inside a component, to show errors on the backend.

I've inspected through the components in the ember-inspector and found out, that the `in` property gets not set to `false`, after the `open` prop was set to `false`.

The modal ~closes fine~ fades out, if I not only overgive in handlebars the `open`-boolean (`open=theBoolThatHandlesThisInTheUpperComp`), but also the `in`-boolean (`open=theBoolThatHandlesThisInTheUpperComp in=theBoolThatHandlesThisInTheUpperComp`). But that ~is just a workaround~ was just a test for now. For sure, the backdrop keeps in the DOM and no interaction is possible. (Yeah, I know it's a private property).

Used versions (I've tried with two ember-cli versions):
- 1.13.12
- 2.9.1

Other versions:

```json
Ember             : 1.13.10
Ember Data        : 2.9.0
jQuery            : 1.11.3
Ember Simple Auth : 1.1.0
```

Update (after an hour): I upgraded ember-cli, ember-data, aso. to the actual version (~2.9), but the error still exists.

### My component in application.hbs

```hbs
{{#backend-error-modal open=openBackendErrorModal}}
  {{backendErrMsg}}
{{/backend-error-modal}}
```

### backend-error-modal comp

```hbs
{{#bs-modal open=open in=open header=false body=false footer=false submitAction=(action "submit") title=title as |modal|}}
  {{#bs-modal-header}}
    <h4 class="modal-title text-danger"><i class="glyphicon glyphicon-{{headerIcon}}"></i> {{title}}</h4>
  {{/bs-modal-header}}
  {{#bs-modal-body}}
    {{yield}}
  {{/bs-modal-body}}
  {{bs-modal-footer closeTitle="Schließen" submitTitle="Zum Login"}}
{{/bs-modal}}
```

```js
import Ember from 'ember';

export default Ember.Component.extend({
  headerIcon: 'comment',
  // Can be overriden via "title" argument on integration of component.
  title: 'Serverfehler',

  actions: {
    close() {
      console.log('Called "close" on backend error modal.');
      this.set('open', false);
    },
    submit() {
      console.log('Called submit on backend error modal.');
      this.set('open', false);
      this.get('router').transitionTo('login');
    }
  }
});
```

### Workaround for the moment

Putting the `open`-Boolean inside an `{{#if ...}}` in the template:

```hbs
{{#if open}}
  {{#bs-modal open=open header=false body=false footer=false submitAction=(action "submit") title=title as |modal|}}
```

... but then, as you know, no animation/transition on toggling the boolean will happen.

**Comment by iamsebastian** [comment]
I've tried all versions between `0.9.1` and `0.10.0`. The *break* happened between `0.9.1` and `0.10.0`.

**Comment by simonihmig** [comment]
I was able to reproduce this in Ember 1.13. In all following versions however I could not find any problem.

I added some changes that should hopefully fix this, and some tests for that use case. I have that in master (https://github.com/kaliber5/ember-bootstrap/commit/f59b94da0b58e3f3f64e9e73c16b2b2fff452038).

Could you please check if that fixes the problem for you? Just replace the ember-bootstrap line in your `package.json` to `"ember-bootstrap": "kaliber5/ember-bootstrap#master"` and run `npm i`.

**Comment by iamsebastian** [comment]
@simonihmig 
Run's fine. Thanks for your work!

**Comment by simonihmig** [comment]
Great, thanks for reporting! This is published right now as 0.11.2
